gui/src/lianalite/login.rs

```diff
@@ -0,0 +1,564 @@
+                            .map_err(|e| Error::Unexpected(e.to_string()))?;
+                        let client = AuthClient::new(
+                            service_config.auth_api_url,
+                            service_config.auth_api_public_key,
+                            auth_config.email,
+                        );
+                        connect_with_refresh_token(
+                            client,
+                            auth_config.refresh_token,
+                            auth_config.wallet_id,
+                            service_config.backend_api_url,
+                        )
+                        .await
+                    },
+                    Message::Connected,
+                ),
+            ),
+        }
+    }
+
+    pub fn stop(&mut self) {}
+
+    pub fn update(&mut self, message: Message) -> Command<Message> {
+        match &mut self.step {
+            ConnectionStep::CheckingAuthFile => {
+                if let Message::Connected(res) = message {
+                    self.processing = false;
+                    match res {
+                        Ok(BackendState::NoWallet(_)) => {
+                            self.auth_error = Some("No wallet found for the given email");
+                        }
+                        Ok(BackendState::WalletExists(client, wallet)) => {
+                            return Command::perform(async move { (client, wallet) }, |(c, w)| {
+                                Message::Run(Ok((c, w)))
+                            });
+                        }
+                        Err(e) => {
+                            self.connection_error = Some(e);
+                            self.step = ConnectionStep::EnterEmail {
+                                email: form::Value::default(),
+                            };
+                        }
+                    }
+                }
+            }
+            ConnectionStep::EnterEmail { email } => match message {
+                Message::View(ViewMessage::EmailEdited(value)) => {
+                    email.valid = value.is_empty() || email_address::EmailAddress::is_valid(&value);
+                    email.value = value;
+                }
+                Message::View(ViewMessage::RequestOTP) => {
+                    if email.value.is_empty() {
+                        email.valid = false;
+                    } else if email.valid {
+                        let email = email.value.clone();
+                        let network = self.network;
+                        self.processing = true;
+                        self.connection_error = None;
+                        self.auth_error = None;
+                        return Command::perform(
+                            async move {
+                                let config = super::client::get_service_config(network)
+                                    .await
+                                    .map_err(|e| Error::Unexpected(e.to_string()))?;
```

Comment:
shouldn't we give a clearer error here? i think it's where we 404 atm on mainnet?